src/lib/mina.ts

```diff
@@ -647,20 +647,42 @@ LocalBlockchain satisfies (...args: any) => Mina;
  * Represents the Mina blockchain running on a real network
  */
 function Network(graphqlEndpoint: string): Mina;
-function Network(graphqlEndpoints: { mina: string; archive: string }): Mina;
-function Network(input: { mina: string; archive: string } | string): Mina {
+function Network(graphqlEndpoints: {
+  mina: string | string[];
+  archive: string | string[];
+}): Mina;
+function Network(
+  input: { mina: string | string[]; archive: string | string[] } | string
+): Mina {
   let accountCreationFee = UInt64.from(defaultAccountCreationFee);
-  let graphqlEndpoint: string;
+  let minaGraphqlEndpoint: string;
   let archiveEndpoint: string;
 
   if (input && typeof input === 'string') {
-    graphqlEndpoint = input;
-    Fetch.setGraphqlEndpoint(graphqlEndpoint);
+    minaGraphqlEndpoint = input;
+    Fetch.setGraphqlEndpoint(minaGraphqlEndpoint);
   } else if (input && typeof input === 'object') {
-    graphqlEndpoint = input.mina;
-    archiveEndpoint = input.archive;
-    Fetch.setGraphqlEndpoint(graphqlEndpoint);
-    Fetch.setArchiveGraphqlEndpoint(archiveEndpoint);
+    if (!input.mina || !input.archive)
+      throw new Error(
+        "Network: malformed input. Please provide an object with 'mina' and 'archive' endpoints."
+      );
+    if (Array.isArray(input.mina) && input.mina.length > 1) {
```

Comment:
I think you should remove the `&& input.mina.length > 1` check here -- there's no reason to fail on an input like `["some-url"]`, it should be treated the same as just a string `"some-url"`